modules/arrow/src/geoarrow/convert-geoarrow-to-binary-geometry.ts

```diff
@@ -136,19 +165,97 @@ function getBinaryGeometriesFromChunk(
       return getBinaryLinesFromChunk(chunk, geoEncoding);
     case 'geoarrow.polygon':
     case 'geoarrow.multipolygon':
-      return getBinaryPolygonsFromChunk(chunk, geoEncoding);
+      return getBinaryPolygonsFromChunk(chunk, geoEncoding, options);
     default:
       throw Error('invalid geoarrow encoding');
   }
 }
 
+/**
+ * get triangle indices. Allows deck.gl to skip performing costly triangulation on main thread.
+ * @param polygonIndices Indices within positions of the start of each simple Polygon
+ * @param primitivePolygonIndices Indices within positions of the start of each primitive Polygon/ring
+ * @param flatCoordinateArray Array of x, y or x, y, z positions
+ * @param nDim - number of dimensions per position
+ * @returns
+ */
+function getTriangleIndices(
+  polygonIndices: Uint16Array,
+  primitivePolygonIndices: Int32Array,
+  flatCoordinateArray: Float64Array,
+  nDim: number
+): Uint32Array {
+  let k = 0;
+  const triangles: number[] = [];
+  // loop polygonIndices to get triangles
+  for (let i = 0; i < polygonIndices.length - 1; i++) {
+    const startIdx = polygonIndices[i];
+    const endIdx = polygonIndices[i + 1];
+    // get subarray of flatCoordinateArray
+    const slicedFlatCoords = flatCoordinateArray.subarray(startIdx * nDim, endIdx * nDim);
+    // get holeIndices for earcut
+    const holeIndices: number[] = [];
+    while (primitivePolygonIndices[k] < endIdx) {
+      if (primitivePolygonIndices[k] > startIdx) {
+        holeIndices.push(primitivePolygonIndices[k] - startIdx);
+      }
+      k++;
+    }
+    const triangleIndices = earcut(slicedFlatCoords, holeIndices, nDim);
```

Comment:
Note to self: OK for now, but 
- I am a little hesitant about adding code to perform earcut into every single geospatial table loader. 
- Ideally we would have one common worker that could perform earcut on the binary output of other loaders. 
- The main problem is that we would need that worker to work on batches - which is a bit fiddly as we will need to build an adapter between asyncIterators and ping-pong worker messaging.

Reply:
Thanks, Ib!  I agree. I think maybe we should do the polygon tesselation for geoarrow in deck.gl: e.g. 
export `function triangulate(polygons: BinaryPolygonFeatures)` in carto-vector-tile-loader (see [here](https://github.com/visgl/deck.gl/blob/b5108c8904797732a83eabb1cd3cdca2f71f6ee4/modules/carto/src/layers/schema/carto-vector-tile-loader.ts#L68)), so we call it in solid-polygon-layer when do polygon tesselating on binary geometry. What do you think?

Reply:
We definitely want to do the tesselation on a worker if we can, and since deck.gl does not really handle async attribute generation yet I think it is reasonable to do it here in loaders.gl. 

I just wish we didn't need to import this code into every loader, but that is something we can optimize later.

Reply:
I haven't looked into it yet, but I was thinking it should be possible for my layers when resolving the `data` input. deck supports that being async because it allows remote file input, so it should be possible to persist earcut's output in the layer state. And maybe also parallelize operations well through a worker farm.

I was thinking of trying to make an isolated helper worker that does earcut on a contiguous polygon array (i.e. my `PolygonData` type, or `arrow.Data<arrow.List<arrow.List<arrow.FixedSizeList<arrow.Float64>>>>`). If you follow the `.data` attributes, you should be able to access the underlying `ArrayBuffer`s and transfer those. Maybe that wouldn't work in a transferable if those ArrayBuffers are shared among multiple columns, but I don't know if ArrowJS does that